colossus-tests/src/test/scala/colossus/service/CallbackSpec.scala

```diff
@@ -310,6 +310,23 @@ class CallbackSpec extends ColossusSpec {
 
     }
 
+    "toFuture" in {
+      import scala.concurrent.ExecutionContext.Implicits.global
+      var value = 0
+      val callback = Callback(func).map { i => i }
+      value = Await.result(callback.toFuture, 2 seconds)
+      value must equal(5)
+    }
+
+    "toFuture failed" in {
+      import scala.concurrent.ExecutionContext.Implicits.global
+
+      val badfunc = {f: (Try[Int] => Unit) => f(Failure[Int](new Exception))}
+      val callback = Callback(badfunc).map { i => i }
+      val result = Await.ready(callback.toFuture, 2 seconds).value.get
+      result mustBe a [Failure[Int]]
```

Comment:
yeah I've run into the same problem before, better to do `result.isFailure must equal(true)`